src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -264,113 +265,120 @@ public Parser(Compilation compilation, in JsonSourceGenerationContext sourceGene
 
                 List<ContextGenerationSpec>? contextGenSpecList = null;
 
-                foreach (ClassDeclarationSyntax classDeclarationSyntax in classDeclarationSyntaxList)
+                foreach (var group in classDeclarationSyntaxList.GroupBy(c => c.SyntaxTree))
```

Comment:
(nit) we only use `var` when it can be inferred from the right-hand side.

See (10) on [https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md)

Reply:
IDE didn't prompt/warn me on this.  Have you set thsi for your editorconfig?

Reply:
We have

https://github.com/dotnet/runtime/blob/20cd393c1e577f0ff53cdc3a00edc3bcfc68ab7f/.editorconfig#L47-L50